### PR TITLE
first routing algo

### PR DESCRIPTION
**Pull request by gstfrenkel**

### CODE REVIEW

File: src/antop.cpp [hunk 1]
```diff
@@ -210,5 +207,43 @@ void Antop::init(const int satellites) {
...
+
+// ToDo:
+// - Test heap.
+// - Update dst to include information about the packet (such as number of hops thus far).
+// - There's a "small" edge case that will break this routing algorithm. Let's say cell A receives a packet and the shortest routes to its destination are through either
+//   cell B or cell C. Cell A then forwards it to cell B, but it turns out that cell B has no option but to return the packet to cell A due to missing satellites. Now,
+//   cell A holds the same packet but lastHop has been updated to cell B. Naturally, it then forwards it to cell C, but cell C experiences the same shortcoming as cell B.
+//   Cell A receives the same packet for a third time, but this time lastHop has been updated to cell C. Therefore, cell A will once again try its luck with cell B, turning
+//   this into an endless loop, until the packet is lost due to the number of hops.
+H3Index Antop::getNextHopId(const H3Index src, const H3Index dst, const H3Index lastHop, bool isDstValid(H3Index idx)) const {
```
Comment (valenadels) [inline src/antop.cpp:221]: que se supone que es el isvaliddst? que deberia pasarle yo desde dtnsim? lo mismo para el lasthop.. no se puede almacenar esto internamente en atop?

File: src/antop.cpp [hunk 2]
```diff
@@ -57,10 +59,25 @@ bool Antop::isNewAddrValid(const Address& addr, const H3Index idx) {
     return true;
 }
 
+void Antop::buildNeighborGraph() {
```
[the remark this answers is not in the document]
Comment (gstfrenkel) [reply src/antop.cpp:62]: Intenté y no encontré la forma

File: src/antop.cpp (hunk 1 above)
Comment (gstfrenkel) [reply src/antop.cpp:221]: El lastHop debería ser parte del paquete. No tiene sentido que lo trackee antop porque pueden haber una banda de paquetes en vuelo.

La función isValidDst es una función que pasa dtnsim para validar si el destino que sugiere ANTop es válido. ANTop puede decir que la mejor ruta es a través de la celda X, pero ANTop no sabe si en esa celda hay algún satélite que pueda recibir el paquete. Entonces lo que hace es llamar a esa función de dtnsim que devuelve un booleano referido a si el destino elegido es válido.

Comment (valenadels) [reply src/antop.cpp:221]: El lastHop debería ser parte del paquete. No tiene sentido que lo trackee antop porque pueden haber una banda de paquetes en vuelo. --> ok

La función isValidDst es una función que pasa dtnsim para validar si el destino que sugiere ANTop es válido.  --> pero no entiendo, es valido si dst == el dst que te paso como destino final

Comment (valenadels) [reply src/antop.cpp:221]: ah no es si es valido el destino, es el nexthop

File: src/antop.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (valenadels) [reply src/antop.cpp:62]: queres q lo veamos?

File: src/antop.cpp (hunk 1 above)
Comment (gstfrenkel) [reply src/antop.cpp:221]: Ok, le actualizo el nombre cuando vuelva a casa.

**Review by valenadels — APPROVED** [review]
